plugins/inputs/nomad/nomad.go

```diff
@@ -0,0 +1,215 @@
+  # auth_token = "/path/to/auth/token"
+  ## OR
+  # auth_token_string = "a1234567-40c7-9048-7bae-378687048181"
+
+  ## Labels to be added as tags. An empty array for both include and
+  ## exclude will include all labels.
+  # label_include = []
+  # label_exclude = ["*"]
+
+  ## Set response_timeout (default 5 seconds)
+  # response_timeout = "5s"
+
+  ## Optional TLS Config
+  # tls_ca = /path/to/cafile
+  # tls_cert = /path/to/certfile
+  # tls_key = /path/to/keyfile
+`
+
+func init() {
+	inputs.Add("nomad", func() telegraf.Input {
+		return &Nomad{
+			LabelInclude: []string{},
+			LabelExclude: []string{"*"},
+		}
+	})
+}
+
+func (n *Nomad) SampleConfig() string {
+	return sampleConfig
+}
+
+func (n *Nomad) Description() string {
+	return "Read metrics from the Nomad api"
+}
+
+func (n *Nomad) Init() error {
+	if n.AuthToken == "" && n.AuthTokenString == "" {
+		n.AuthToken = ""
+	}
+
+	if n.AuthToken != "" {
+		token, err := os.ReadFile(n.AuthToken)
+		if err != nil {
+			return err
+		}
+		n.AuthTokenString = strings.TrimSpace(string(token))
+	}
+
+	labelFilter, err := filter.NewIncludeExcludeFilter(n.LabelInclude, n.LabelExclude)
+	if err != nil {
+		return err
+	}
+	n.labelFilter = labelFilter
+
+	return nil
+}
+
+func (n *Nomad) Gather(acc telegraf.Accumulator) error {
+	acc.AddError(n.gatherSummary(n.URL, acc))
+	return nil
+}
+
+func (n *Nomad) gatherSummary(baseURL string, acc telegraf.Accumulator) error {
+	summaryMetrics := &MetricsSummary{}
+	err := n.LoadJSON(fmt.Sprintf("%s/v1/metrics", baseURL), summaryMetrics)
+	if err != nil {
+		return err
+	}
+
+	buildNomadMetrics(summaryMetrics, acc)
+
+	return nil
+}
+
+func (n *Nomad) LoadJSON(url string, v interface{}) error {
+	var req, err = http.NewRequest("GET", url, nil)
+	if err != nil {
+		return err
+	}
+	var resp *http.Response
+	tlsCfg, err := n.ClientConfig.TLSConfig()
+	if err != nil {
+		return err
+	}
+	if n.RoundTripper == nil {
+		if n.ResponseTimeout < config.Duration(time.Second) {
+			n.ResponseTimeout = config.Duration(time.Second * 5)
+		}
+		n.RoundTripper = &http.Transport{
+			TLSHandshakeTimeout:   5 * time.Second,
+			TLSClientConfig:       tlsCfg,
+			ResponseHeaderTimeout: time.Duration(n.ResponseTimeout),
+		}
+	}
+	req.Header.Set("Authorization", "X-Nomad-Token "+n.AuthTokenString)
+	req.Header.Add("Accept", "application/json")
+	resp, err = n.RoundTripper.RoundTrip(req)
+	if err != nil {
+		return fmt.Errorf("error making HTTP request to %s: %s", url, err)
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode != http.StatusOK {
+		return fmt.Errorf("%s returned HTTP status %s", url, resp.Status)
+	}
+
+	err = json.NewDecoder(resp.Body).Decode(v)
+	if err != nil {
+		return fmt.Errorf(`error parsing response: %s`, err)
+	}
+
+	return nil
+}
+
+func buildNomadMetrics(summaryMetrics *MetricsSummary, acc telegraf.Accumulator) {
+
+	t, err := time.Parse(timeLayout, summaryMetrics.Timestamp)
+	if err != nil {
+		panic(err)
+	}
```

Comment:
No. Simply return the error here. If your plugin crashes for any reason it will now rip-down all telegraf which is not good as all other configured plugins still work as expected.